### PR TITLE
Implement microprofile reactive messaging

### LINKED ISSUE #483: Implement an EventService using MP reactive messaging
Microprofile provides a generic message processing system with [reactive streams](https://github.com/eclipse/microprofile-reactive-messaging/blob/master/spec/src/main/asciidoc/architecture.asciidoc). There is a more complete description on the [SmallRye website](https://smallrye.io/smallrye-reactive-messaging/).

This would likely be an easy way to integrate an `EventService` component into a microprofile-based application while also being generic across message brokers.

### PR DESCRIPTION
**Pull request by acoburn**
Resolves #483

This is still a bit experimental, but it passes the tests and there is so little code here, that I am inclined to move this forward.

### CODE REVIEW

File: notifications/reactive/src/test/java/org/trellisldp/reactive/ReactiveEventServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        service.emit(mockEvent);
+        await().atMost(FIVE_SECONDS).until(() -> collector.getResults().size() == 1);
+        service.emit(mockEvent);
+        await().atMost(FIVE_SECONDS).until(() -> collector.getResults().size() == 2);
+        service.emit(mockEvent);
+        service.emit(mockEvent);
+        await().atMost(FIVE_SECONDS).until(() -> collector.getResults().size() == 4);
+    }
+
+    private <T> T getInstance(final Class<T> type) {
```
Comment (ajs6f) [inline notifications/reactive/src/test/java/org/trellisldp/reactive/ReactiveEventServiceTest.java]: If the idea is to get properly set-up beans from the container, can we not just inject them? I'm not at all familiar with Weld's test gear so maybe not?

Comment (acoburn) [reply notifications/reactive/src/test/java/org/trellisldp/reactive/ReactiveEventServiceTest.java]: I'm also not super familiar with Weld's test gear, but I'll see if there is something easy that will do that injection for us.

Comment (ajs6f) [reply notifications/reactive/src/test/java/org/trellisldp/reactive/ReactiveEventServiceTest.java]: https://github.com/weld/weld-junit/tree/master/junit5

Comment (acoburn) [reply notifications/reactive/src/test/java/org/trellisldp/reactive/ReactiveEventServiceTest.java]: That last commit should address this. It also makes the test code much more compact.

File: notifications/reactive/src/main/java/org/trellisldp/reactive/ReactiveEventService.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+@ApplicationScoped
+public class ReactiveEventService implements EventService {
+
+    public static final String REACTIVE_DESTINATION = "trellis";
+
+    private final ActivityStreamService serializer;
+    private final PublishSubject<Message<String>> subject = PublishSubject.create();
+
+    /**
+     * Create a new Kafka Event Service.
```
Comment (ajs6f) [inline notifications/reactive/src/main/java/org/trellisldp/reactive/ReactiveEventService.java]: Is this "Kafka" left over from some cut-and-paste?

Comment (acoburn) [reply notifications/reactive/src/main/java/org/trellisldp/reactive/ReactiveEventService.java]: that's now fixed.

**Pull request merged**